### PR TITLE
feat: [IDP-1408]: Support for GitHub security Data Points

### PR DESCRIPTION
**Pull request by vigneswara-propelo**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54257)

### CODE REVIEW

File: idp-service/src/main/resources/migrations/scorecard/checks.json [hunk 1]
```diff
@@ -509,5 +509,130 @@
     "tags": ["Software quality", "Development maturity"],
     "defaultBehaviour": "FAIL",
     "failMessage": "Number of workflows is less than or equal to 3."
+  },
+  {
+    "id": "652e8e9116ff017907f1bc64",
+    "accountIdentifier": "__GLOBAL_ACCOUNT_ID__",
+    "identifier": "__HARNESS__864af47a-cb3b-7d98-ada7-9478ab7d6a5e",
+    "name": "Number of open critical alerts reported by dependabot is less than 3",
```
Comment (OrkoHunter) [inline idp-service/src/main/resources/migrations/scorecard/checks.json:517]:
```suggestion
    "name": "Open Dependabot alerts is less than 3",
  
    
```
IMO the names should be shorter and descriptions have more details.

Comment (OrkoHunter) [reply idp-service/src/main/resources/migrations/scorecard/checks.json:517]: Also please use the correct case when writing a Product/Brand/Feature name. `Dependabot`.

File: idp-service/src/main/resources/migrations/scorecard/datasourceLocations.json [hunk 2]
```diff
@@ -381,6 +381,71 @@
         "Authorization": "{BEARER_AUTH}"
       }
     }
+  },
+  {
+    "id": "652d0aff299e6ea7ffe52784",
+    "accountIdentifier": "__GLOBAL_ACCOUNT_ID__",
+    "identifier": "github_open_dependabot_alerts",
```
Comment (sathish-soundarapandian) [inline idp-service/src/main/resources/migrations/scorecard/datasourceLocations.json:388]: If we see these 3 below DSLs, just one or two params change in the URL. I will see if we can club it together.

File: idp-service/src/main/resources/migrations/scorecard/checks.json (hunk 1 above)
Comment (vigneswara-propelo) [reply idp-service/src/main/resources/migrations/scorecard/checks.json]: Done

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 5 of 14 files at r1, 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54257)* status: 14 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`idp-service/src/main/resources/migrations/scorecard/datasourceLocations.json` line 388 at r1](https://reviewable.io/reviews/harness/harness-core/54257#gh-1362307721:-Nh-CflN0xUEPHCgrOVt:b-h5ccqp) ([raw file](https://github.com/harness/harness-core/blob/86bbb2689ca662f8ab7578f3fa0c68d3a646d672/idp-service/src/main/resources/migrations/scorecard/datasourceLocations.json#L388)):*
<details><summary><i>Previously, sathish-soundarapandian (Sathish) wrote…</i></summary><blockquote>

If we see these 3 below DSLs, just one or two params change in the URL. I will see if we can club it together.
</blockquote></details>

Cool. This PR doesn't have to wait for it right? Intelliflo might be waiting for these checks

**Review by vikyathharekal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54257)* status: 14 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`idp-service/src/main/resources/migrations/scorecard/datasourceLocations.json` line 388 at r1](https://reviewable.io/reviews/harness/harness-core/54257#gh-1362307721:-Nh-ElcKAA1h_iT1i-YJ:bdqqlq5) ([raw file](https://github.com/harness/harness-core/blob/86bbb2689ca662f8ab7578f3fa0c68d3a646d672/idp-service/src/main/resources/migrations/scorecard/datasourceLocations.json#L388)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Cool. This PR doesn't have to wait for it right? Intelliflo might be waiting for these checks
</blockquote></details>

If you are already working on it then we could have this as well for the next release. 
The changes we discussed (moving the base URL and token to the DS layer) could wait if it's taking time. We don't need it urgently

**Comment by vigneswara-propelo** [comment]
trigger uts

**Comment by vigneswara-propelo** [comment]
trigger ut9

**Review by sathish-soundarapandian — APPROVED** [review]

**Comment by vigneswara-propelo** [comment]
trigger review

**Pull request merged**